sdk/spring/spring-cloud-azure-appconfiguration-config/src/main/java/com/azure/spring/cloud/appconfiguration/config/implementation/AppConfigurationPropertySource.java

```diff
@@ -49,12 +92,227 @@ public Object getProperty(String name) {
+                featureConfigurationSettings.add(setting);
+                FeatureFlagConfigurationSetting featureFlag = (FeatureFlagConfigurationSetting) setting;
+
+                String configName =
+                    FEATURE_MANAGEMENT_KEY + setting.getKey().trim().substring(FEATURE_FLAG_PREFIX.length());
+
+                updateTelemetry(featureFlag, tracing);
+
+                properties.put(configName, createFeature(featureFlag));
+            } else if (StringUtils.hasText(setting.getContentType())
+                && JsonConfigurationParser.isJsonContentType(setting.getContentType())) {
+                Map<String, Object> jsonSettings = JsonConfigurationParser.parseJsonSetting(setting);
+                for (Entry<String, Object> jsonSetting : jsonSettings.entrySet()) {
+                    key = trimKey(jsonSetting.getKey(), trimStrings);
+                    properties.put(key, jsonSetting.getValue());
+                }
+            } else {
+                properties.put(key, setting.getValue());
+            }
+        }
+    }
+
+    protected String trimKey(String key, List<String> trimStrings) {
+        key = key.trim();
+        if (trimStrings != null) {
+            for (String trim : trimStrings) {
+                if (key.startsWith(trim)) {
+                    return key.replaceFirst("^" + trim, "").replace('/', '.');
+                }
+            }
+        }
+        return key.replace("/", ".");
+    }
+
+    /**
+     * Given a Setting's Key Vault Reference stored in the Settings value, it will get its entry in Key
+     * Vault.
+     *
+     * @param secretReference {"uri": "&lt;your-vault-url&gt;/secret/&lt;secret&gt;/&lt;version&gt;"}
+     * @return Key Vault Secret Value
+     */
+    protected String getKeyVaultEntry(AppConfigurationKeyVaultClientFactory keyVaultClientFactory,
+        SecretReferenceConfigurationSetting secretReference) {
+        String secretValue = null;
+        try {
+            URI uri = null;
+            KeyVaultSecret secret = null;
+
+            // Parsing Key Vault Reference for URI
+            try {
+                uri = new URI(secretReference.getSecretId());
+                secret = keyVaultClientFactory.getClient("https://" + uri.getHost()).getSecret(uri, maxRetryTime);
+            } catch (URISyntaxException e) {
+                LOGGER.error("Error Processing Key Vault Entry URI.");
+                ReflectionUtils.rethrowRuntimeException(e);
+            }
+
+            if (secret == null) {
+                throw new IOException("No Key Vault Secret found for Reference.");
+            }
+            secretValue = secret.getValue();
+        } catch (RuntimeException | IOException e) {
+            LOGGER.error("Error Retrieving Key Vault Entry");
+            ReflectionUtils.rethrowRuntimeException(e);
+        }
+        return secretValue;
+    }
+
+    private String getFeatureSimpleName(ConfigurationSetting setting) {
```

Comment:
Same here, the feature related method should only be in the feature related impl.

Reply:
Can't as snapshots also uses it.